pkg/template/resource.go

```diff
@@ -46,40 +47,61 @@ type getClusterTriggerBinding func(name string, options metav1.GetOptions) (*tri
 // ResolveTrigger takes in a trigger containing object refs to bindings and
 // templates and resolves them to their underlying values.
 func ResolveTrigger(trigger triggersv1.EventListenerTrigger, getTB getTriggerBinding, getCTB getClusterTriggerBinding, getTT getTriggerTemplate) (ResolvedTrigger, error) {
-	tb := make([]*triggersv1.TriggerBinding, 0, len(trigger.Bindings))
-	ctb := make([]*triggersv1.ClusterTriggerBinding, 0, len(trigger.Bindings))
-	for _, b := range trigger.Bindings {
-		if b.Spec != nil {
-			tb = append(tb, &triggersv1.TriggerBinding{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: b.Name,
-				},
-				Spec: *b.Spec,
+	bp, err := ResolveBindingsToParams(trigger.Bindings, getTB, getCTB)
+	if err != nil {
+		return ResolvedTrigger{}, fmt.Errorf("failed to resolve bindings: %w", err)
+	}
+	if trigger.Template.Name == "" {
+		return ResolvedTrigger{}, fmt.Errorf("trigger template name is empty")
+	}
+	tt, err := getTT(trigger.Template.Name, metav1.GetOptions{})
+	if err != nil {
+		return ResolvedTrigger{}, fmt.Errorf("error getting TriggerTemplate %s: %w", trigger.Template.Name, err)
+	}
+	return ResolvedTrigger{TriggerTemplate: tt, BindingParams: bp}, nil
+}
+
+// ResolveBindingsToParams takes in both embedded bindings and references and returns a list of resolved Param values.ResolveBindingsToParams
+func ResolveBindingsToParams(bindings []*triggersv1.TriggerSpecBinding, getTB getTriggerBinding, getCTB getClusterTriggerBinding) ([]triggersv1.Param, error) {
```

Comment:
Doesn't look like this is being used outside of this package. Let's unexport this for now until we have a clear need, particularly since this relies on unexported types as parameters.

Reply:
done